File: sample/app-java/src/main/java/com/appice/sample/services/AppICEFirebaseListener.java
package com.appice.sample.services;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import semusi.activitysdk.ContextSdk;

public class AppICEFirebaseListener extends FirebaseMessagingService {
    @Override
    public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);
        System.out.println("onMessage : "+remoteMessage.getData());
        boolean isAppICEPush=  ContextSdk.isAppICENotification(remoteMessage);
        if (isAppICEPush){
            ContextSdk.handleAppICEPush(remoteMessage, getApplicationContext());
        }
    }
}

File: sample/app-kotlin/src/main/java/com/semusi/services/AppICEFirebaseListener.kt
package com.semusi.services

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import semusi.activitysdk.ContextSdk

class AppICEFirebaseListener : FirebaseMessagingService() {
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        println("onMessage : " + remoteMessage.data)
        val isAppICEPush = ContextSdk.isAppICENotification(remoteMessage)
        if (isAppICEPush) {
            ContextSdk.handleAppICEPush(remoteMessage, this.applicationContext)
        }
    }
}